Clients/GUI/Pages/SignUpWizzard.razor

```diff
@@ -0,0 +1,126 @@
+                    <InputText class="form-control col-3" type="password" @bind-Value="userInput.Password" placeholder="*****" />
+                </div>
+
+                <div class="form-group">
+                    <label for="ConfirmPassword">Confirm Password</label>
+                    <InputText class="form-control col-3" type="password" @bind-Value="userInput.ConfirmPassword" placeholder="*****" />
+                </div>
+                <br />
+
+                <ValidationSummary />
+                @if (isErrorDisplayed)
+                {
+                    <label style=" color: red;"> @errorMessage </label>
+                    <br />
+                }
+
+            </WizardStep>
+            <WizardStep Name="Second Step">
+                <div class="row">
+                    <div class="container m-5">
+                        <label>Send your avatar</label>
+                        <br />
+                        <InputFile OnChange="HandleSelection" />
+                        <br />
+                        @if (isErrorDisplayed)
+                        {
+                            <label style="color: red;"> @errorMessage </label>
+                            <br />
+                        }
+                        <ValidationSummary />
+                    </div>
+                </div>
+            </WizardStep>
+        </Wizard>
+    </EditForm>
+
+@code {
+    bool isErrorDisplayed = false;
+    string errorMessage = "";
+    private SignUpViewModel userInput;
+
+    public async Task TrySignUp()
+    {
+        try
+        {
+            var request = new CreateUserRequest
+            {
+                FirstName = userInput.FirstName,
+                LastName = userInput.LastName,
+                Birthday = userInput.Birthday,
+                Email = userInput.Email,
+                Password = userInput.Password,
+                AvatarExtension = userInput.TypeAvatar,
+                Avatar = userInput.Avatar
+            };
+
+            await SignUpper.SignUp(request);
+            NavigationManager.NavigateTo("/signin");
+        }
+        catch (BadRequestException)
+        {
+            userInput = new SignUpViewModel();
+            isErrorDisplayed = true;
+            errorMessage = "Something went wrong during registration";
+        }
+        catch (NotFoundException)
+        {
+            userInput = new SignUpViewModel();
+            isErrorDisplayed = true;
+            errorMessage = "Something went wrong during signing you in";
+        }
+        catch (Exception e)
+        {
+            userInput = new SignUpViewModel();
+            isErrorDisplayed = true;
+            errorMessage = e.Message;
+        }
+    }
+
+    protected override void OnInitialized()
+    {
+        userInput = new SignUpViewModel();
+    }
+
+    async Task HandleSelection(IFileListEntry[] files)
+    {
+        var file = files.FirstOrDefault();
```

Comment:
Do we suppose to check size of image  in order to don't send 4 GB images?

Reply:
fixed